security/advancedtls/provider.go

```diff
@@ -0,0 +1,199 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package advancedtls
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"google.golang.org/grpc/credentials/tls/certprovider"
+	"google.golang.org/grpc/grpclog"
+)
+
+const (
+	defaultInterval = 1 * time.Hour
+)
+
+var logger = grpclog.Component("channelz")
+
+// IdentityPemFileProviderOptions contains fields to be filled out by users
+// for obtaining identity private key and certificates from PEM files.
+type IdentityPemFileProviderOptions struct {
+	CertFile string
+	KeyFile  string
+	Interval time.Duration
```

Comment:
The interval field needs to be documented as well. It should be clarified that the user can leave this unspecified and if left unspecified what is the default value that will be used?